--- bin/automake~	Fri Feb 13 07:17:13 1998
+++ bin/automake	Sun Feb 15 05:18:39 1998
@@ -882,7 +882,7 @@
 	    $base = $1;
 	    ($hname = $2) =~ tr/y/h/;
 	    ($cname = $2) =~ tr/y/c/;
-	    $output_rules .= "${base}.${hname}: ${base}.${cname}\n";
+	    $output_rules .= "\$(srcdir)/${base}.${hname}: \$(srcdir)/${base}.${cname}\n";
 	}
 	$output_rules .= "\n";
 
@@ -2723,13 +2723,13 @@
 	    {
 		# The dependency points to the current directory in
 		# some way.
-		($xform = $one_dep) =~ s/^$fixup_rx//;
+		($xform = $one_dep) =~ s/^$fixup_rx/\$(srcdir)\//;
 		push (@dependencies, $xform);
 	    }
 	    elsif ($one_dep =~ /^$srcdir_rx/)
 	    {
 		# The dependency is in some other directory in the package.
-		($xform = $one_dep) =~ s/^$srcdir_rx/$rewrite_builddir/;
+		($xform = $one_dep) =~ s/^$srcdir_rx/\$(top_srcdir)\//;
 		push (@dependencies, $xform);
 	    }
 	    elsif ($one_dep =~ /^\//)
@@ -2765,21 +2765,17 @@
 	    $output_rules .= &file_contents ('depend');
 	    push (@clean, 'depend');
 	    &push_phony_cleaners ('depend');
-	    # FIXME: should use @EXT@ and @PFX@.  I don't know why it
-	    # didn't work.
 	    $output_rules .=
-		&file_contents_with_transform ('s/EXT/.c/g;'
-					       . 's/PFX//g;',
+		&file_contents_with_transform ('s/\@EXT\@/.c/g;'
+					       . 's/\@PFX\@//g;',
 					       'depend2');
 	    local ($ext);
 	    local ($need_cxx) = 0;
 	    foreach $ext (sort keys %cxx_extensions)
 	    {
-		# FIXME: should use @EXT@ and @PFX@.  I don't know why
-		# it didn't work.
 		$output_rules .=
-		    &file_contents_with_transform ('s/EXT/' . $ext .'/g;'
-						   . 's/PFX/CXX/g;',
+		    &file_contents_with_transform ('s/\@EXT\@/' . $ext .'/g;'
+						   . 's/\@PFX\@/CXX/g;',
 						   'depend2');
 		$need_cxx = 1;
 	    }
@@ -2793,8 +2789,7 @@
     {
 	# Include any auto-generated deps that are present.  Note that
 	# $build_directory ends in a "/".
-	if (-d ($build_directory . $relative_dir . "/.deps")
-	    && -f ($build_directory . $relative_dir . "/.deps/.P"))
+	if (-d ($build_directory . $relative_dir . "/.deps"))
 	{
 	    local ($depfile);
 
@@ -2896,8 +2891,8 @@
     {
 	local (@ac_deps) = (
 			    ($seen_maint_mode ? "\@MAINT\@" : "") ,
-			    "configure.in",
-			    ($acinclude ? ' acinclude.m4' : '')
+			    '$(top_srcdir)/configure.in',
+			    ($acinclude ? ' $(srcdir)/acinclude.m4' : '')
 			    );
 
 	# Scan all -I directories for m4 files.  These are our
@@ -2991,7 +2986,7 @@
 		      # Don't know why.
 		      . ': '
 		      . ($seen_maint_mode ? '@MAINT@ ' : '')
-		      . $amfile . ' '
+		      . '$(srcdir)/' . $amfile . ' '
 		      . '$(top_srcdir)/configure.in $(ACLOCAL_M4) '
 		      . join (' ', @rewritten) . "\n"
 		      . "\tcd \$(top_srcdir) && \$(AUTOMAKE) "
@@ -3053,21 +3048,24 @@
 					  $FOREIGN, $ch_sans_dir);
 
 	    # Header defined and in this directory.
-	    local (@files);
+	    local (@files, @relfiles);
 	    if (-f $relative_dir . '/acconfig.h')
 	    {
-		push (@files, 'acconfig.h');
+		push (@relfiles, "acconfig.h");
+		push (@files, "\\\$(srcdir)/acconfig.h");
 	    }
 	    if (-f $one_name . '.top')
 	    {
-		push (@files, "${cn_sans_dir}.top");
+		push (@relfiles, "${cn_sans_dir}.top");
+		push (@files, "\\\$(srcdir)/${cn_sans_dir}.top");
 	    }
 	    if (-f $one_name . '.bot')
 	    {
-		push (@files, "${cn_sans_dir}.bot");
+		push (@relfiles, "${cn_sans_dir}.bot");
+		push (@files, "\\\$(srcdir)/${cn_sans_dir}.bot");
 	    }
 
-	    &push_dist_common (@files);
+	    &push_dist_common (@relfiles);
 
 	    local ($stamp_name) = 'stamp-h';
 	    $stamp_name .= "${hdr_index}" if scalar (@config_headers) > 1;
--- share/automake/depend2.am~	Fri Feb 13 06:33:04 1998
+++ share/automake/depend2.am	Sun Feb 15 04:42:53 1998
@@ -15,14 +15,24 @@
 ## along with this program; if not, write to the Free Software
 ## Foundation, Inc., 59 Temple Place - Suite 330, Boston, MA
 ## 02111-1307, USA.
-## FIXME: should use @EXT@, but that didn't work.
-%.o: %EXT
-## FIXME: should use @PFX@, but that didn't work.
-	@echo '$(PFXCOMPILE) -c $<'; \
+%.o: %@EXT@
+	@echo '$(@PFX@COMPILE) -c $<'; \
 	DEPENDENCIES_OUTPUT='.deps/$(*F).P'; \
 	export DEPENDENCIES_OUTPUT; \
 ## Note that using DEPENDENCIES_OUTPUT causes gcc to append to the
 ## named file.  So we truncate it explicitly.
 	: > .deps/$(*F).P; \
-## FIXME: should use @PFX@, but that didn't work.
-	$(PFXCOMPILE) -c $<
+	$(@PFX@COMPILE) -c $<
+
+%.lo: %@EXT@
+	@echo '$(LT@PFX@COMPILE) -c $<'; \
+	DEPENDENCIES_OUTPUT='.deps/$(*F).p'; \
+	export DEPENDENCIES_OUTPUT; \
+## Note that using DEPENDENCIES_OUTPUT causes gcc to append to the
+## named file.  So we truncate it explicitly.
+	: > .deps/$(*F).p; \
+	$(LT@PFX@COMPILE) -c $<
+	@-sed -e 's/^\([^:]*\)\.o:/\1.lo \1.o:/' < .deps/$(*F).p | \
+	$(AWK) '/:/ { ignore = ++ignoring[$$1]; } ignore == 1 { print }' \
+	> .deps/$(*F).P
+	@-rm -f .deps/$(*F).p
