--- autoreconf	Wed Nov 27 00:39:57 1996
+++ autoreconf	Mon Jul  7 06:43:39 1997
@@ -19,12 +19,15 @@
 
 usage="\
 Usage: autoreconf [-f] [-h] [--help] [-m dir] [--macrodir=dir]
-       [-l dir] [--localdir=dir] [--force] [--verbose] [--version]"
+       [-l dir] [--localdir=dir] [--force] [--verbose] [--version]
+       [--cygnus] [--foreign] [--gnits] [--gnu] [-i] [--include-deps]"
 
 localdir=
 verbose=no
 show_version=no
 force=no
+automake_mode=--gnu
+automake_deps=
 
 test -z "$AC_MACRODIR" && AC_MACRODIR=/n/gnu/autoconf-2.12/share/autoconf
 
@@ -54,6 +57,10 @@
     force=yes; shift ;;
   --version | --vers*)
     show_version=yes; shift ;;
+  --cygnus | --foreign | --gnits | --gnu)
+    automake_mode=$1; shift ;;
+  --include-deps | -i)
+    automake_deps=$1; shift ;;
   --)     # Stop option processing.
     shift; break ;;
   -*) echo "$usage" 1>&2; exit 1 ;;
@@ -111,6 +118,44 @@
        aclocal=$dots$localdir/aclocal.m4 ;;
   esac
 
+  # Regenerate aclocal.m4 if necessary.  FIXME: if aclocal searches
+  # nonstandard directories, we need to deal with that here.  The
+  # easiest way is to move this info into configure.in.
+  run_aclocal=no
+  if test -f "$aclocal" &&
+     grep 'generated automatically by aclocal' $aclocal > /dev/null
+  then
+     run_aclocal=yes
+  else
+     if test -f `dirname $aclocal`/acinclude.m4
+     then
+	run_aclocal=yes
+     fi
+  fi
+  if test $run_aclocal = yes
+  then
+     if test $force = no &&
+        ls -lt configure.in $aclocal \
+	       `dirname $aclocal`/acinclude.m4 | sed 1q |
+          grep 'aclocal.m4$' > /dev/null
+     then
+	:
+     else
+	test $verbose = yes && echo running aclocal in $dir, creating $aclocal
+	aclocal --output=$aclocal -I `dirname $aclocal`
+     fi
+  fi
+
+  # Re-run automake if required.  Assumes that there is a Makefile.am
+  # in the topmost directory.
+  if test -f Makefile.am
+  then
+     amforce=
+     test $force = no && amforce=--no-force
+     test $verbose = yes && echo running automake in $dir
+     automake $amforce $automake_mode $automake_deps
+  fi
+
   test ! -f $aclocal && aclocal=
 
   if test $force = no && test -f configure &&
@@ -123,27 +168,35 @@
     $autoconf $macrodir_opt $localdir_opt
   fi
 
-  if grep AC_CONFIG_HEADER configure.in >/dev/null; then
-    template=`sed -n '/AC_CONFIG_HEADER/{
-s%[^#]*AC_CONFIG_HEADER(\([^)]*\).*%\1%
+  if grep 'A[CM]_CONFIG_HEADER' configure.in >/dev/null; then
+    templates=`sed -n '/A[CM]_CONFIG_HEADER/{
+s%[^#]*A[CM]_CONFIG_HEADER[ 	]*(\([^)]*\).*%\1%
 t here
 : here
-s%.*:%%
-t colon
-s%$%.in%
-: colon
 p
 q
 }' configure.in`
-    if test ! -f $template || grep autoheader $template >/dev/null; then
+    tcount=`set -- $templates; echo $#`
+    template=`set -- $templates; echo $1 | sed '
+s/.*://
+t there
+s/$/.in/
+: there
+'`
+    stamp=`dirname $template`/stamp-h`test "$tcount" -gt 1 && echo 1`.in
+    if test ! -f "$template" || grep autoheader "$template" >/dev/null; then
       if test $force = no && test -f $template &&
-	ls -lt $template configure.in $aclocal | sed 1q |
-	  grep "$template$" > /dev/null
+	 ls -lt $template configure.in $aclocal 2>/dev/null \
+	        `echo $localdir_opt | sed 's/--localdir=//
+		                           s%\(.\)$%\1/%'`acconfig.h |
+	   sed 1q | grep "$template$" > /dev/null
       then
         :
       else
         test $verbose = yes && echo running autoheader in $dir
-        $autoheader $macrodir_opt $localdir_opt
+        $autoheader $macrodir_opt $localdir_opt && 
+	{ test $verbose != yes || echo touching $stamp; } &&
+	touch $stamp
       fi
     fi
   fi
--- aclocal	Wed Jun 25 16:32:02 1997
+++ aclocal	Mon Jul  7 05:12:58 1997
@@ -377,8 +377,8 @@
 
     print STDERR "Writing aclocal.m4\n" if $verbosity;
 
-    open (ACLOCAL, "> aclocal.m4")
-	|| die "aclocal: couldn't open \`aclocal.m4' for writing: $!\n";
+    open (ACLOCAL, "> " . $output_file)
+	|| die "aclocal: couldn't open \`" . $output_file . "' for writing: $!\n";
     print ACLOCAL "dnl aclocal.m4 generated automatically by aclocal $VERSION\n\n";
     print ACLOCAL $output;
     close (ACLOCAL);
